flashinfer/deep_gemm/m_grouped_gemm.py

```diff
@@ -0,0 +1,430 @@
+"""
```

Comment:
Keep the original license first.

Also, the naming convention "m_" do not align with flashinfer's, we can just call it "grouped_gemm".